### ISSUE TITLE
Remote node debugging don't work in this situation.

### ISSUE THREAD

**Issue by chansey97** [ignatov/intellij-erlang#768]
I met a problem when I using intellij-erlang plugin to debug ejabberd.

Reproduce step:
1. Download ejabberd windows release installer and install it.
2. Download ejabberd source code and compile it by rebar.
Note: since many ejabberd dependencies (like: zlib iconv expat) are not included in its source code, I can only compile it with "skip-deps".
3. Copy compiled beams to ejabberd release install directory and replace original beams. 
4. Import ejabberd source code to intellij-erlang IDE .
5. Add a "Erlang Remote Node" configuration: set module, nodename, cookie but remove Make configuration at "Before launch: Another Configuration, Active tool window".
6. Start ejabberd server.
7. Run this configuration in intellij-erlang IDE. (success! no error occured!)
8. Set a breakpoint at a socket data receive function, for example: ejabberd_router : route
9. Start xmpp client to connect server.
10. The whole application hangs and no any prompt in intellij-erlang IDE...

Note: Use the same method, but use erlang official debugger can be successfully debug.

**Comment by deadok22** [comment]
Hey @chansey97,

Could you enable debugger logs as described in #512 and share the logs?

> 11. The whole application hangs and no any prompt in intellij-erlang IDE...

Which application hangs? If it's the debugged app, could you try generating a crash dump and look for suspicious processes.

**Comment by chansey97** [comment]
Hi @deadok22 
The hanged application is ejabberd server not intellij-erlang IDE.
I uploaded log.
[log_server_hang.txt](https://github.com/ignatov/intellij-erlang/files/504831/log_server_hang.txt)

**Comment by deadok22** [comment]
@chansey97 the log looks normal: ```int:ni``` is called successfully, the breakpoint is set, and nothing happens afterwards. 

So could you try generating a crash dump for the ejabberd process and try to figure out what is going on by looking at process' stacks?

**Comment by chansey97** [comment]
@deadok22 
The ejabberd server does not crash , it maybe only stuck at break point... (so no crash dump, no error log)
I'm newer to erlang, can you give me some suggestion about how to generating a crash dump ?

**Comment by deadok22** [comment]
@chansey97 call ```erlang:halt(1)```

**Comment by chansey97** [comment]
[erl_crash_20161002-230311.zip](https://github.com/ignatov/intellij-erlang/files/505049/erl_crash_20161002-230311.zip)

@deadok22 Upload crash dump.

**Comment by deadok22** [comment]
@chansey97 try to identify the process that's supposed to be running but is stuck. You also may want to look at crash dump of the debugger node (the process spawned by the plugin to attach to your process). Also, look for suspicious cpu/disk activity spikes.

**Comment by chansey97** [comment]
Yes, but this may need a lot of effort....
Is it possible that your plugin has received the message from the server but not process?
Since official debugger works normally.

**Comment by deadok22** [comment]
> Is it possible that your plugin has received the message from the server but not process?
I don't think so, but I can't be 100% sure there's no bugs. Here's the [debugger code](https://github.com/deadok22/erlang-debug-node/blob/master/src/remote_debugger_notifier.erl). As you can see, it's basically forwarding messages to the IDE. 

> And...how to generate crash dump of the debugger node (the process spawned by the plugin to attach to your process).
The same way as any other erlang process: connect to it and execute the code you need. Look at how the debugger does that [here](https://github.com/deadok22/erlang-debug-node/blob/master/src/remote_debugger_listener.erl).

**Comment by chansey97** [comment]
@deadok22 
Is it possible that there is inconsistent path caused?
For example:
windows path: is \\
linux path: is /
This cause plugin could not find source code.

**Comment by chansey97** [comment]
Hi @deadok22,
I found a phenomenon may be related to this issue.

Using your plugin, I could debug ejabberd at some lines of source code line now, but at other lines of source code it will crash (no dumpfile generated).
Use the official debugger of erlang, it also have some opportunity to crash (no dumpfile generated).

But when I set "Stack On, No Tail" to "Stack Off" in official debugger, the problem resolved.

So I think it might be a stack overflow problem, but I don't know why.

Is there any way to set stack property for intellij-erlang plugin "Stack On, No Tail" to "Stack Off" like the official debugger?

**Comment by deadok22** [comment]
@chansey97 try calling ```int:stack_trace(false)``` in the vm being debugged: http://erlang.org/doc/man/int.html#stack_trace-1

Although, I still wonder why the default ```no_tail``` doesn't suit you. Is it because you run out of memory?

**Comment by chansey97** [comment]
@deadok22 I called int:stack_trace(false) at ejabberd node shell and use your plugin to remote debug.
But the problem still unresolved.
The different between intellij-erlang plugin debugger and official debugger is:
intellij-erlang plugin debugger using remote debugging
official debugger using local debugging

**Comment by deadok22** [comment]
Could you try calling that on the debugger node then?

**Comment by chansey97** [comment]
@deadok22 I'm newer to inteillij, can you tell me how to calling that on the debugger node? (Need recompile intellij-erlang plugin source code?)
There is no input shell when tarting intellij-erlang plugin debugger.
